apps/comps/rpc/middleware/cache.ts

```diff
@@ -0,0 +1,70 @@
+import { StandardRPCJsonSerializer } from "@orpc/client/standard";
+import {
+  Context,
+  ErrorMap,
+  Meta,
+  Middleware,
+  ORPCErrorConstructorMap,
+  os,
+} from "@orpc/server";
+import { unstable_cache } from "next/cache";
+import type { Logger } from "pino";
+
+export function cacheMiddleware<
```

Comment:
this could benefit from some doc comments, its relatively complex and non necessary obvious what all the args and generics represent